library/alloc/src/iter.rs

```diff
@@ -0,0 +1,38 @@
+//! Allocation extensions for [`Iterator`].
+//!
+//! *[See also the Iterator trait][Iterator].*
+#![unstable(feature = "iterator_join", issue = "75638")]
+
+use crate::slice::Join;
+use crate::vec::Vec;
+
+/// Iterator extension traits that requires allocation.
+#[unstable(feature = "iterator_join", issue = "75638")]
+pub trait IteratorExt: Iterator {
```

Comment:
The `impl` seems to be missing

Reply:
Do we need to `impl IteratorExt` for all items that `impl Iterator`? I thought this already does that?

Reply:
Without any `impl`, it's just specifying the supertrait relationship as a requirement to implementors. I would recommend a blanket `impl<T: Iterator + ?Sized>`, which will include the `dyn` case.